File: src/main/webapp/src/app/types/filter/framework/filter-settings-framework.ts
import { ConditionFieldName } from './condition-field-name'
import { ConditionAmountParameter } from './condition-amount-parameter'
import { ConditionTitleParameter } from './condition-title-parameter'
import { ConditionDateParameter } from './date-parameter'
import { ConditionMatchingOption } from './condition-matching-option'

export interface FilterSettingsFramework {
    conditionFieldNames: ConditionFieldName[]
    conditionAmountParameters: ConditionAmountParameter[]
    conditionTitleParameters: ConditionTitleParameter[]
    conditionDateParameters: ConditionDateParameter[]
    conditionMatchingOptions: ConditionMatchingOption[]
}

File: src/main/java/com/example/filter/test/model/filter/framework/FilterSettingsFramework.java
package com.example.filter.test.model.filter.framework;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.util.List;

@Getter
@AllArgsConstructor
public class FilterSettingsFramework {
	List<ConditionFieldName> conditionFieldNames;
	List<ConditionAmountParameter> conditionAmountParameters;
	List<ConditionTitleParameter> conditionTitleParameters;
	List<ConditionDateParameter> conditionDateParameters;
	List<ConditionMatchingOption> conditionMatchingOptions;
}
